diff --git a/Marlin/Configuration.h b/Marlin/Configuration.h
index f02e69e..17a1f53 100644
--- a/Marlin/Configuration.h
+++ b/Marlin/Configuration.h
@@ -104,7 +104,7 @@
  *
  * :[-1, 0, 1, 2, 3, 4, 5, 6, 7]
  */
-#define SERIAL_PORT 0
+#define SERIAL_PORT 1
 
 /**
  * Select a secondary serial port on the board to use for communication with the host.
@@ -128,7 +128,7 @@
 
 // Choose the name from boards.h that matches your setup
 #ifndef MOTHERBOARD
-  #define MOTHERBOARD BOARD_RAMPS_14_EFB
+  #define MOTHERBOARD BOARD_STM32F103RE
 #endif
 
 // Name displayed in the LCD "Ready" message and Info menu
diff --git a/platformio.ini b/platformio.ini
index 9e41afa..5c9cbf7 100644
--- a/platformio.ini
+++ b/platformio.ini
@@ -29,7 +29,6 @@ lib_deps =
   TMCStepper@>=0.6.2,<1.0.0
   Adafruit NeoPixel
   U8glib-HAL=https://github.com/MarlinFirmware/U8glib-HAL/archive/bugfix.zip
-  Adafruit_MAX31865=https://github.com/adafruit/Adafruit_MAX31865/archive/master.zip
   LiquidTWI2=https://github.com/lincomatic/LiquidTWI2/archive/master.zip
   Arduino-L6470=https://github.com/ameyer/Arduino-L6470/archive/0.8.0.zip
   SailfishLCD=https://github.com/mikeshub/SailfishLCD/archive/master.zip
